test/fixtures/azure-fixture/src/main/java/fixture/azure/MockAzureBlobStore.java

```diff
@@ -0,0 +1,483 @@
+
+        void releaseLease(String leaseId);
+
+        void assertCanRead(@Nullable String leaseId);
+
+        void assertCanWrite(@Nullable String leaseId);
+
+        boolean isCommitted();
+    }
+
+    private abstract static class AbstractAzureBlob implements AzureBlob {
+
+        protected final Object writeLock = new Object();
+        protected final Lease lease = new Lease();
+
+        @Override
+        public String acquireLease(@Nullable String proposedLeaseId, int leaseTimeSeconds) {
+            synchronized (writeLock) {
+                return lease.acquire(proposedLeaseId, leaseTimeSeconds);
+            }
+        }
+
+        @Override
+        public void releaseLease(String leaseId) {
+            synchronized (writeLock) {
+                lease.release(leaseId);
+            }
+        }
+
+        @Override
+        public void assertCanRead(@Nullable String leaseId) {
+            lease.assertCanRead(leaseId);
+        }
+
+        @Override
+        public void assertCanWrite(@Nullable String leaseId) {
+            lease.assertCanWrite(leaseId);
+        }
+    }
+
+    private static class MockAzureBlockBlob extends AbstractAzureBlob implements AzureBlob {
+        private final Map<String, BytesReference> blocks;
+        private volatile BytesReference contents = EMPTY;
+
+        private MockAzureBlockBlob() {
+            this.blocks = new ConcurrentHashMap<>();
+        }
+
+        public void putBlock(String blockId, BytesReference content, @Nullable String leaseId) {
+            synchronized (writeLock) {
+                lease.assertCanWrite(leaseId);
+                this.blocks.put(blockId, content);
+            }
+        }
+
+        public void putBlockList(List<String> blockIds, @Nullable String leaseId) throws BadRequestException {
+            synchronized (writeLock) {
+                lease.assertCanWrite(leaseId);
+                final List<String> unresolvedBlocks = blockIds.stream().filter(bId -> blocks.containsKey(bId) == false).toList();
+                if (unresolvedBlocks.isEmpty() == false) {
+                    logger.warn("Block list contained non-existent block IDs: {}", unresolvedBlocks);
+                    throw new BadRequestException("InvalidBlockList", "The specified blocklist is invalid.");
+                }
+                final BytesReference[] resolvedContents = blockIds.stream().map(blocks::get).toList().toArray(new BytesReference[0]);
+                contents = CompositeBytesReference.of(resolvedContents);
+            }
+        }
+
+        private boolean matches(String ifNoneMatchHeaderValue) {
+            if (ifNoneMatchHeaderValue == null) {
+                return false;
+            }
+            // We only support *
+            if ("*".equals(ifNoneMatchHeaderValue)) {
+                return true;
+            }
+            throw new BadRequestException("UnsupportedHeader", "The test fixture only supports * for if-none-match");
```

Comment:
Let's fail the test directly here.

Reply:
Done in f37dc830f8a5c60fe0f17ce4f67640bc9fac186f